### PR TITLE
CD pipeline execution walkthrough

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [x ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-76823

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by thisrohangupta — DISMISSED** [review]

File: docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+<docimage path={require('./static/77e84ac51371ea84e703528c89ea1ce75dc2d3e13a634a3a0e7ed350d0d95562.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+Here's the process:
+
+<docimage path={require('./static/3c43b3302b31c37f0c91e826d42f7e86bcfa17f3aff68f35831eaf5a48887b19.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+#### Artifact configuration (optional)
+
+You can add an artifact to the service and use the artifact expression `image: <+artifacts.primary.image>` in your values YAML file to fetch the image tag information and pass it into the manifest.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md:184]:
```suggestion
You can add an artifact to the service and use the artifact expression `image: <+artifacts.primary.image>` in your `values.yaml` file to fetch the image tag information and pass it into the manifest.
```

File: docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+<docimage path={require('./static/9ea6e614fb430935b293312a635e6365de5c5110f54247313e56e6e42e744fff.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+Here's the process:
+
+<docimage path={require('./static/896f6ecff8cc8c8889043fbb5656484b607be65b12e56800119035482f881024.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+#### Variables (optional)
+
+You can configure variables in a service and reference them in the service's values YAML file or in pipeline execution steps.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md:206]:
```suggestion
You can configure variables in a service and reference them in the service's `values.yaml` file or in pipeline execution steps.
```

File: docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+The next entities evaluated during pipeline execution are the stage environment and infrastructure definition.
+
+<docimage path={require('./static/b052e20f319cb8d113b8cdf8001832eb7629f1002568d5d5595aeb426467c774.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+
+A Harness environment defines **where** the service will be deployed in terms of Production and Pre-Production environments. Within an environment, you can have multiple infrastructure definitions, and you select the one to use for each pipeline stage.
+
+Each infrastructure definition defines the specific infrastructure where the pipeline will deploy. For example, the cloud platform or Kubernetes cluster.
+
+The infrastructure definition for Kubernetes (including Helm and Kustomize) and Native Helm also contain a **Release Name**. The release name is used to track the deployed service and is used to sync instances so users know how many pods are deployed on a cluster.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md:223]:
```suggestion
The infrastructure definition for Kubernetes (including Helm and Kustomize) and Native Helm also contain a **Release Name**. The release name is used to track the deployed service and to sync instances so users know how many pods are deployed on a cluster.
```

File: docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+<docimage path={require('./static/ffac7e13dca40a4c238eb1bb1b2762654e05b190093f10b354f80178a8d5225d.png')} width="60%" height="60%" title="Click to view full size image" />  
+
+When the Rollout Deployment step is executed, the delegate eligible to perform the Kubernetes Rollout Deployment step is assigned the task. Harness evaluates delegates in a round robin style.
+
+The Rollout Deployment step is made up of 2 tasks that Harness provides out of the box:
+
+1. Fetch files task.
+   1. The fetch files task leverages the GitHub connector configured in the service to fetch the manifests.
+   2. Harness will also render and resolve any of the Harness variables defined in the values.yaml of the service and add them to the manifest/Helm chart using Go/Helm templating.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md:254]:
```suggestion
   2. Harness will also render and resolve any of the Harness variables defined in the `values.yaml` file of the service and add them to the manifest/Helm chart using Go/Helm templating.
```

File: docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+   3. Harness fetches any secrets referenced in the values.yaml and resolves them in the manifest. Harness masks secret output in its logs.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/get-started/pipeline-execution-walkthrough.md:255]:
```suggestion
   3. Harness fetches any secrets referenced in the `values.yaml` file and resolves them in the manifest. Harness masks secret output in its logs.
```

**Review by brian-f-harness — APPROVED** [review]
LGTM. May consider removing the left nav from screenshots because of the upcoming changes.

**Pull request merged**